src/gui/menu_filesystem.cpp

```diff
@@ -98,7 +102,13 @@ FileSystemMenu::refresh_items()
 
   for (const auto& item : m_files)
   {
-    add_entry(item_id, item);
+    ItemAction& entry = add_entry(item_id, item);
+    std::string help_text = m_generate_help_text_for_file ? m_generate_help_text_for_file(FileSystem::join(m_directory, item)) : "";
+
+    if (!help_text.empty()) {
+      entry.set_help(help_text);
+    }
```

Comment:
Redundant `.empty()` check.

Reply:
Thanks for the feedback. I don't see how this condition could be removed and also avoid calling entry.set_help(string) when string is empty. The string is empty when m_generate_help_text_for_file is nullptr (not defined) or if m_generate_help_text_for_file(string) returns the empty string, which could happen in the event of a failure to load the file or when it finds no data for a file. We already have a conditional for the former condition on the previous line, but the !help_text.empty() is needed to handle the later condition. Are you recommending that we call set_help(string) when string is empty?

Reply:
Yes, wouldn't the item just not have any help text set if `string` is empty?

Reply:
You're right, MenuItem::set_help(string) behaves well even in the empty string case. I pushed a commit that implements your suggested change. Thanks!